UnityProject/Assets/Scripts/Player/Huggable.cs

```diff
@@ -82,8 +96,46 @@ private bool TryFieryHug()
 		return false;
 	}
 
+	private void Judgement(LivingHealthMasterBase puller)
+	{
+		if (lastPuller == null || lastPuller != puller)
+		{
+			JudgementCooldown(puller);
+			return;
+		}
+		if (DMMath.Prob(tailPullJudgementChance))
+		{
+			puller.Gib();
+		}
+	}
+
+	private async void JudgementCooldown(LivingHealthMasterBase puller)
+	{
+		lastPuller = puller;
+		await Task.Delay(tailPullJudgementCooldown).ConfigureAwait(false);
+		lastPuller = null;
+	}
+
+	private bool PullTail()
+	{
+		var performerLHB = interaction.Performer.GetComponent<LivingHealthMasterBase>();
+		var targetLHB = interaction.TargetObject.GetComponent<LivingHealthMasterBase>();
+		if (targetLHB == null) return false;
+		foreach (var possibleTail in targetLHB.BodyPartList)
+		{
+			if (possibleTail.name.Contains("Tail") == false) continue;
```

Comment:
This seems like a rather snowflakey way to check if something is a tail or not. It would be a good idea to classify tail parts somehow rather than just doing a string comparison.

Reply:
I already did try comparing objects but that didn't work

Reply:
`if (tails.Contains(possibleTail.gameObject) == false) continue;` could i be doing this wrong?